plugins/dfu/dfu-target.h

```diff
@@ -84,6 +65,38 @@ typedef enum {
 	DFU_TARGET_TRANSFER_FLAG_LAST
 } DfuTargetTransferFlags;
 
+struct _DfuTargetClass
+{
+	GUsbDeviceClass		 parent_class;
+	void			 (*percentage_changed)	(DfuTarget	*target,
+							 guint		 percentage);
+	void			 (*action_changed)	(DfuTarget	*target,
+							 FwupdStatus	 action);
+	gboolean		 (*setup)		(DfuTarget	*target,
+							 GCancellable	*cancellable,
+							 GError		**error);
+	gboolean		 (*attach)		(DfuTarget	*target,
+							 GCancellable	*cancellable,
+							 GError		**error);
+	gboolean		 (*detach)		(DfuTarget	*target,
+							 GCancellable	*cancellable,
+							 GError		**error);
+	gboolean		 (*mass_erase)		(DfuTarget	*target,
+							 GCancellable	*cancellable,
+							 GError		**error);
+	DfuElement		*(*upload_element)	(DfuTarget	*target,
+							 guint32	 address,
+							 gsize		 expected_size,
+							 gsize		 maximum_size,
+							 GCancellable	*cancellable,
+							 GError		**error);
+	gboolean		 (*download_element)	(DfuTarget	*target,
+							 DfuElement	*element,
+							 DfuTargetTransferFlags flags,
+							 GCancellable	*cancellable,
+							 GError		**error);
+};
```

Comment:
This may not be a large issue, and intended but i'll mention in case it's an oversight.
You previously had padding for 8 reservations but only added 6 function assignments.

Reply:
Good point, but I removed the padding as we're not installing the library I don't care about ABI much. If you're more comfortable I can remove all the class padding in a separate commit.

Reply:
No worry, I think it's fine as is.  just wanted to make sure you were aware.  Since it's internal private ABI probably not a big deal.

Reply:
I've removed this manually in master.